src/Test/Inspection/Plugin.hs

```diff
@@ -318,14 +414,27 @@ resultToExpr (Failure s) = App <$> dcExpr 'Failure <*> mkStringExpr s
 proofPass :: UponFailure -> ReportingMode -> ModGuts -> CoreM ModGuts
 proofPass upon_failure report guts = do
     dflags <- getDynFlags
-    let noopt = optLevel dflags < 1
+    let noopt =
+#if MIN_VERSION_ghc(9,4,0)
+            snd $ updOptLevelChanged 1 dflags
```

Comment:
It's not unheard to do `-fno-full-laziness`. Won't this be a `False` then, though enough optimizations are on for `inspection-testing` to work.

I hope this regression is fixed before GHC-9.4 is released, so we won't need to hack around it.

Reply:
Right, but this discussion shows that trying to guess whether the user intended to run the tests or not is not very helpful in the first place.